src/PointwiseFunctions/GeneralRelativity/WeylMagnetic.hpp

```diff
@@ -0,0 +1,119 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace gsl {
+template <typename>
+struct not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace gr {
+
+//@(
+/*!
+ * \ingroup GeneralRelativityGroup
+ * \brief Computes the magnetic part of the Weyl tensor.
+ *
+ * \details Computes the magnetic part of the Weyl tensor \f$B_{ij}\f$
+ * as: \f$ B_{ij} =
+ * \left(1/\sqrt{\det\gamma}\right)D_{k}K_{l(i}\g_{j)m}\epsilon^{mlk}
+ * \f$ where \f$\epsilon^{ijk}\f$ is the spatial Levi-Civita symbol,
+ * \f$K_{ij}\f$
+ * is the extrinsic curvature, \f$\g_{jm} \f$ is the spatial metric,
+ * and \f$D_i\f$ is spatial covariant derivative.
+ */
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::ii<DataType, SpatialDim, Frame> weyl_magnetic(
+    const tnsr::ijj<DataType, SpatialDim, Frame>& grad_extrinsic_curvature,
+    const tnsr::ii<DataType, SpatialDim, Frame>& spatial_metric) noexcept;
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void weyl_magnetic(
+    gsl::not_null<tnsr::ii<DataType, SpatialDim, Frame>*> weyl_magnetic_part,
+    const tnsr::ijj<DataType, SpatialDim, Frame>& grad_extrinsic_curvature,
+    const tnsr::ii<DataType, SpatialDim, Frame>& spatial_metric) noexcept;
+//@}
+
+// @{
+/*!
+ * \ingroup GeneralRelativityGroup
+ * \brief Computes the scalar \f$B_{ij} B^{ij}\f$ from the magnetic
+ * part of the Weyl tensor \f$B_{ij}\f$.
+ *
+ * \details Computes the scalar \f$B_{ij} B^{ij}\f$ from the magnetic part
+ * of the Weyl tensor \f$B_{ij}\f$ and the inverse spatial metric
+ * \f$g^{ij}\f$, i.e. \f$B_{ij} = \g^{ik}\g^{jl}B_{ij}B_{kl}\f$.
```

Comment:
`g` is used for the spacetime metric, we use `\gamma` for the spatial metric. Please apply this change throughout the PR :)